### PR TITLE
WiP: Edu Enrollments Schema

### PR DESCRIPTION
**Pull request by omgitsbillryan**
Connects: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2922

This also completes the following:
- Updates model & serializer to [latest EVSS spec](https://github.com/department-of-veterans-affairs/vets.gov-team/blob/master/Products/EVSS%20Integration/Post-911%20Enrollment%20Status/education_enrollment_schema.jpg)

### CODE REVIEW

File: app/controllers/v0/education_enrollments_controller.rb [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# frozen_string_literal: true
+module V0
+  class EducationEnrollmentsController < ApplicationController
+    def show
+      # TODO: fetch for realz
+      ee = FactoryGirl.build(:education_enrollment)
```
Comment (omgitsbillryan) [inline app/controllers/v0/education_enrollments_controller.rb]: Please do not merge until we address this line.  We either need to comment it out or wait until we are able to fetch real data from EVSS.

File: spec/support/schema_matchers.rb [hunk 2]
```diff
@@ -4,17 +4,17 @@
 module SchemaMatchers
   extend RSpec::Matchers::DSL
 
-  def valid_against_schema?(json, schema_name)
+  def valid_against_schema?(json, schema_name, opts = {})
     schema_path = Rails.root.join('spec', 'support', 'schemas', "#{schema_name}.json")
-    JSON::Validator.validate!(schema_path.to_s, json, strict: true)
+    JSON::Validator.validate!(schema_path.to_s, json, { strict: true }.merge(opts))
```
Comment (omgitsbillryan) [inline spec/support/schema_matchers.rb]: The EVSS API allows for missing key:value pairs.  `strict: true` basically disallows this so I had to make a way to override it.

File: spec/support/schemas/education_enrollment.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "$schema" : "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "required": ["data"],
```
Comment (kreek) [inline spec/support/schemas/education_enrollment.json]: As EVSS could be down but vets-api up, we could add a meta object that has server status like https://github.com/department-of-veterans-affairs/vets-api/blob/0874ca62f189a7751b923809df75f52c2d2935f9/spec/support/schemas/letters.json#L64

**Comment by ayaleloehr** [comment]
Calling all of these education_enrollment feels off to me, as this product is not enrolling someone in education. Since Vets.gov is doing so many things (including applying for education benefits which also sounds similar to enrollment), I think accurately naming each product and having the directory and file names accurately describe what that product is doing is super important. Is there a better way to name these, either in this or a separate PR?

cc @robbiethegeek-usds

**Comment by omgitsbillryan** [comment]
@ayaleloehr I completely agree.  It was originally called `education_enrollment_status` but that was just too long and cluttering up code making it unreadable and file names too long.

Beyond the class name, I've had a really tough time figuring out what to call this: "Post 911 GI Bill", "Post 911 Education Enrollment", "Education Enrollment Status", "Post 911 Status" to name a few.

**Comment by omgitsbillryan** [comment]
The front-end is calling it `Post911GIBStatus`, I think it may be a winner

**Comment by omgitsbillryan** [comment]
Closing this PR since [this commit](https://github.com/department-of-veterans-affairs/vets-api/commit/cf8fb24e3dff3a4d1f4e55434c6e5d2b53d205d3) was a branch off of this one and contains all code in this PR.

**Pull request closed without merging**